File: Data Structures & Algorithms/remove-duplicates-from-sorted-array/submission-0.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        l, r = 0, 1
        while r < len(nums):
            while r < len(nums) and nums[l] == nums[r]:
                r += 1
            if r >= len(nums):
                break
            nums[l + 1] = nums[r]
            l += 1
            r += 1
        return l + 1

        

File: Data Structures & Algorithms/binary-search/submission-0.java
class Solution {
    private int binarySearch(int l, int r, int[] nums, int target) {
        if (l > r) return -1;

        int m = (l + r) / 2;

        if (nums[m] == target) return m;
        if (target > nums[m]) {
            return binarySearch(m + 1, r, nums, target);
        } else {
            return binarySearch(l, m - 1, nums, target);
        }
    }

    public int search(int[] nums, int target) {
        return binarySearch(0, nums.length - 1, nums, target);
    }
}
